packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js

```diff
@@ -40,38 +39,46 @@ angular.module('App').controller(
       };
 
       this.model = {
-        capacity: null,
+        ram: null,
         duration: null,
         contract: false,
       };
 
       this.database = this.$scope.currentActionData;
 
-      this.$scope.sortRam = (ram) => +ram;
+      this.$scope.sortRam = (element) => +element.capacity;
 
       this.userService.getUser().then((user) => {
         this.data.ovhSubsidiary = user.ovhSubsidiary;
       });
 
       /*= =============================
-             =            STEP 1            =
-             ============================== */
+       =            STEP 1            =
+       ============================== */
       this.loading.availableRam = true;
 
       this.privateDatabaseService
         .getUpgradePlans(this.productId)
         .then((result) => {
-          this.data.availableRam = result.map(
-            (e) => e.planCode.match(/[0-9]+/)?.[0],
-          );
-          if (this.database.infrastructure === 'legacy') {
-            remove(this.data.availableRam, (ram) => ram === '2048');
-          }
-
-          remove(
-            this.data.availableRam,
-            (ram) => +ram === +this.database.ram.value,
-          );
+          angular.forEach(result, (e) => {
```

Comment:
Can we retain map here ? result.map()

Reply:
@darsene suggested something else above. Not sure which one of the suggestion I should apply.